src/Illuminate/Http/Middleware/Http2ServerPush.php

```diff
@@ -0,0 +1,135 @@
+<?php
+
+namespace Illuminate\Http\Middleware;
+
+use Closure;
+use Illuminate\Http\Request;
+use Illuminate\Http\Response;
+use Symfony\Component\DomCrawler\Crawler;
+
+class Http2ServerPush
+{
+    /**
+     * The DomCrawler instance.
+     *
+     * @var \Symfony\Component\DomCrawler\Crawler
+     */
+    protected $crawler;
+
+    /**
+     * Handle an incoming request.
+     *
+     * @param \Illuminate\Http\Request $request
+     * @param \Closure $next
+     *
+     * @return mixed
+     */
+    public function handle(Request $request, Closure $next, $limit = null)
+    {
+        $response = $next($request);
+
+        if ($response->isRedirection() || ! $response instanceof Response || $request->isJson()) {
```

Comment:
Does it really matter if `$request->isJson()`? I imagine that it's important what the response is, not what the request that generated the response contains.

Reply:
Also, the order of checks. Should you check for the Response type before calling `->isRedirection()`?

Reply:
Great points here. Re-ordered the checks, and now checking the response for JSON instead of the `request`